src/firefly/js/core/ComponentCntlr.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import {flux} from '../Firefly.js';
+import {get} from 'lodash';
+import update from 'react-addons-update';
+
+export const DIALOG_OR_COMPONENT_KEY = 'dialogOrComponent';
+
+const DIALOG_KEY = 'dialog';
+const COMPONENT_KEY = 'component';
+
+/*---------------------------- ACTIONS -----------------------------*/
+
+const SHOW_DIALOG = 'ComponentCntlr.showDialog';
+const HIDE_DIALOG = 'ComponentCntlr.hideDialog';
+const HIDE_ALL_DIALOGS = 'ComponentCntlr.hideAllDialogs';
+const COMPONENT_STATE_CHANGE = 'ComponentCntlr.componentStateChange';
+
+
+
+export default {
+    SHOW_DIALOG,
+    HIDE_DIALOG,
+    COMPONENT_STATE_CHANGE,
+    reducer
+};
+
+//======================================== Dispatch Functions =============================
+//======================================== Dispatch Functions =============================
+//======================================== Dispatch Functions =============================
+
+
+export function dispatchShowDialog(dialogId,ownerId=undefined) {
+    flux.process({type: SHOW_DIALOG, payload: {dialogId, ownerId}});
+}
+
+export function dispatchHideDialog(dialogId) {
+    flux.process({type: HIDE_DIALOG, payload: {dialogId}});
+}
+
+export function dispatchHideAllDialogs() {
+    flux.process({type: HIDE_ALL_DIALOGS, payload: {}});
+}
+
+export function dispatchComponentStateChange(componentId, componentState) {
+    flux.process({type: COMPONENT_STATE_CHANGE, payload: {componentId, componentState}});
+}
+
+
+//============================ Utilities ============================
+//============================ Utilities ============================
+//============================ Utilities ============================
+
+export function isDialogVisible(dialogKey) {
+    return get(flux.getState()[DIALOG_OR_COMPONENT_KEY], [DIALOG_KEY,dialogKey,'visible'],false);
+}
+
+export function getDialogOwner(dialogKey) {
+    const d= get(flux.getState()[DIALOG_OR_COMPONENT_KEY], [DIALOG_KEY,dialogKey], {});
+    return (d.visible && d.ownerId) ? d.ownerId : null;
+}
+
+export function getComponentState(componentKey) {
+    return get(flux.getState()[DIALOG_OR_COMPONENT_KEY], [COMPONENT_KEY,componentKey], {});
+}
+
+/*
+   This controller handles small (one level deep) state of various application components,
+   such as dialogs, tab and collapsible panels, which are not a part of field group
+
+   Structure of the store:
+   dialogOrComponent:
+     dialog:
+        [dialogId]: {
+            visible
+            ownerId
+        }
+     component:
+        [componentId]: savedComponentState
+ */
```

Comment:
I like this `structure of the store` section.  I think we all have been doing this different ways.  We might think about standardizing it.